skbio/io/format/phylip.py

```diff
@@ -208,12 +208,43 @@
                         unicode_literals)
 
 from skbio.alignment import Alignment
+from skbio.sequence import Sequence
 from skbio.io import create_format, PhylipFormatError
 from skbio.util._misc import chunk_str
 
+
 phylip = create_format('phylip')
 
 
+@phylip.sniffer()
+def _phylip_sniffer(fh):
+    # Strategy:
+    #   Read the header and a single sequence; verify that the sequence length
+    #   matches the header information.  Do not verify that the total number of
+    #   lines matches the header information, since that would require reading
+    #   the whole file.
+    try:
+        header = next(_line_generator(fh))
+    except StopIteration:
+        return False, {}
+    try:
+        _, seq_len = _validate_header(header)
+    except PhylipFormatError:
+        return False, {}
+    line = next(_line_generator(fh))
+    try:
+        _validate_line(line, seq_len)
+    except PhylipFormatError:
+        return False, {}
+    return True, {}
+
+
+@phylip.reader(Alignment)
+def _phylip_to_alignment(fh):
```

Comment:
Can you add a constructor argument which can be used?

`_phylip_to_alignment(fh, constructor=Sequence)`